src/solvers/dg/dg_1d.jl

```diff
@@ -22,6 +22,12 @@ function create_cache(mesh::TreeMesh{1}, equations::AbstractEquations{1},
   cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg, uEltype)...)
   cache = (;cache..., create_cache(mesh, equations, dg.mortar, uEltype)...)
 
+  # TODO: This should be moved somewhere else but that would require more involved
+  #       changes since we wouldn't know `nelements(dg, cache)`...
+  du_ec  = Array{uEltype, 3}(undef, nvariables(equations), nnodes(dg), nelements(dg, cache))
+  du_cen = similar(du_ec)
+  cache = (;cache..., du_ec, du_cen)
```

Comment:
It would (somewhat) violate the calling convention, but could we just pass `nelements(dg, cache)` or `cache` to the corresponding `create_cache` method? Just a thought, I don't think this requires fixing right now, since we might switch to element-wise kernels anyways.